Add typings to package.json

xiamx opened the pull request:
This is related to https://github.com/vitaly-t/pg-promise/pull/238 and https://github.com/vitaly-t/pg-promise/pull/232#issuecomment-260197804

xiamx committed:
Add typings to package.json

blakeembrey commented:
Just FYI, the file should be written in external module format for TypeScript. To do that, remove the `declare module` from around the definition (https://github.com/vitaly-t/spex/blob/master/typescript/spex.d.ts). Also, I'd kill the `@types` directory in there, I have no idea why that's there.

xiamx committed:
More update for typescript patch

xiamx committed:
include tyepscript as devdependency

xiamx commented:
I updated this patch to included the changes suggested by @blakeembrey; added a test case for some rudimentary typescript compilation check; and finally updated the README under `typescript/` to reflect changes made in this patch.

vitaly-t commented:
@xiamx After accepting the PR, I've made a few changes, including the deletion of folder `@types`.

Now with all the changes related to building local test files, I'm having trouble to figure out how to do that. It worked previously with the references and a complete command-line, both of which you removed.

But now if I run `build.bat`, I'm getting:

```
batch.ts(1,26): error TS2307: Cannot find module 'spex'.
```

What am I missing?

I did run `typings install`, though it didn't do anything, since you removed file `typings.json`.

So how did you test it?

vitaly-t commented:
@xiamx 

Ok, for now I have set relative path in all the test files, so that seems to work, but I'm not sure that this is the right way to do it...

Example:

``` js
import * as spexLib from '../../typescript/spex';

var spex = spexLib(Promise);

function source() {

}

type PageError = typeof spex.errors.PageError;

spex.page(source)
    .then((data: spexLib.TPageResult) => {
        var p: number = data.pages;
    })
    .catch((error: PageError) => {
        var duration: number = error.duration;
    });
```

@blakeembrey is this the right way to do it or there is a better way (maybe specifying path in `tsconfig.json` - I'm just guessing)?

blakeembrey commented:
To be completely honest with you, I'm not sure TypeScript 100% considered this part of the tooling. With Typings, you can do `typings bundle` to generate a definition. Also, with TypeScript 2.0 you can use `typeRoots` to specify where things load. However, for testing your own definition, you basically need to do relative imports, yes. You _may_ be able to abuse `npm link` inside itself, but I haven't tested that.

xiamx commented:
Abusing npm link does work :) I think that's how it's being done by travisCI

vitaly-t commented:
@xiamx ok, this one is now released, will start looking at `pg-promise`...
